pandas/tests/indexes/common.py

```diff
@@ -667,19 +667,19 @@ def test_astype_category(self, copy, name, ordered, simple_index):
         # standard categories
         dtype = CategoricalDtype(ordered=ordered)
         result = idx.astype(dtype, copy=copy)
-        expected = CategoricalIndex(idx.values, name=name, ordered=ordered)
+        expected = CategoricalIndex(idx, name=name, ordered=ordered)
```

Comment:
maybe have a test explicitly for RangeIndex?

Reply:
`tests.indexes.ranges.test_range.py::TestRangeIndex` inherits from this base class, so it's tested that way. Do you have something else in mind?

Reply:
Some tm.assert_foo functions dont raise when we assert an Int64Index equals a RangeIndex.  So off the top my head it isn't obvious that this edited test would fail in master.  im asking for a test that is super-obviously tied to this bugfix